### ISSUE TITLE
[Bug] Formatting Toolbar: If use the context wrapper it. In formattingToolbar can't access the context value.

### ISSUE THREAD

**Issue by ParseDark** [TypeCellOS/BlockNote#189]
```
// index.js
const editor: BlockNoteEditor | null = useBlockNote({
    customElements: {
      formattingToolbar: FormattingToolbar,
    },
  });


// FormattingToolbar
const ImproveIt = (props: { editor: BlockNoteEditor }) => {
  // const [, { pushSelfItem }] = useChatHistoryContext();
  const a = useChatHistoryContext();
  console.log(a); // {}

  return (
    <ActionBox
      title="Improve Writing"
      tooltipMsg="Improve Writing"
      icon={<GrOptimize className="text-gray-600 mr-1" />}
      isSelected={
        props.editor.getActiveStyles().textColor === "blue" &&
        props.editor.getActiveStyles().backgroundColor === "blue"
      }
      onClick={() => {
        const selection = props.editor.getSelection();
        if (selection) {
          const allSelectionText = getSelectionAllText(selection);
          pushSelfItem(`${promote}${allSelectionText}`);
        }
      }}
    />
  );
};
```

You can see the a = {}. I think it's a bug for the Formatting Toolbar.

**Comment by ParseDark** [comment]
Update: I try to use this deps?: DependencyList params. But not work.

<img width="634" alt="image" src="https://user-images.githubusercontent.com/26950370/236818045-f774eb8b-d16d-4ce8-820e-286dcc16c82c.png">

<img width="719" alt="image" src="https://user-images.githubusercontent.com/26950370/236818522-5e1bc977-2460-45fd-8f6c-e8df5d4e0ee3.png">

I expect that in the `Formatting Toolbar` component I can access the `useChatHistoryContext`. Then I can send a message to the chatbox.

**Comment by ParseDark** [comment]
### how to re-produce
```
// import logo from './logo.svg'
import "@blocknote/core/style.css";
import { BlockNoteView, useBlockNote, BlockNoteEditor } from "@blocknote/react";
import { FC, createContext, useContext, useState } from "react";
import { Toolbar } from "../../../packages/react/src/SharedComponents/Toolbar/components/Toolbar";
import { ToolbarButton } from "../../../packages/react/src/SharedComponents/Toolbar/components/ToolbarButton";

// context level
const DemoContext = createContext({});

const useDemo = () => {
  const [v, setV] = useState([]);

  return {
    v,
    setV,
  };
};

type CompType = () => JSX.Element;

// context provider
const withChatHistoryContext = (Comp: CompType) => {
  const NewComp: FC<any> = (props) => {
    const context = useDemo();
    return (
      <DemoContext.Provider value={context}>
        <Comp {...props} />
      </DemoContext.Provider>
    );
  };

  return NewComp;
};

// custom formattingToolbar

const CustomFormattingToolbar = (props: { editor: BlockNoteEditor }) => {
  const useDemoContext = useContext(DemoContext);

  return (
    <Toolbar>
      <ToolbarButton
        mainTooltip={"Blue Text & Background"}
        onClick={() => {
          props.editor.toggleStyles({
            textColor: "blue",
            backgroundColor: "blue",
          });
          debugger;
          console.log(useDemoContext);
        }}
        isSelected={
          props.editor.getActiveStyles().textColor === "blue" &&
          props.editor.getActiveStyles().backgroundColor === "blue"
        }>
        Blue
      </ToolbarButton>
    </Toolbar>
  );
};

type WindowWithProseMirror = Window & typeof globalThis & { ProseMirror: any };

function App() {
  const { v, setV } = useContext(DemoContext) as any;

  console.log(v);

  const editor = useBlockNote(
    {
      customElements: {
        formattingToolbar: CustomFormattingToolbar,
      },
    },
    [v]
  );

  // Give tests a way to get prosemirror instance
  (window as WindowWithProseMirror).ProseMirror = editor?._tiptapEditor;

  return (
    <>
      <button onClick={() => setV((p: any) => [...p, 1])}>click</button>
      <BlockNoteView editor={editor} />
    </>
  );
}

export default withChatHistoryContext(App);
```

So in this example `CustomFormattingToolbar` can't get the `useContext(DemoContext) ` real value. Will always equal "{}".

**Comment by YousefED** [comment]
Thanks @ParseDark. This seems to be an architectural issue indeed. We'll look into it

**Comment by matthewlipski** [comment]
@ParseDark This issue should be fixed in 0.8.3, can you check this?
